gym/envs/tests/test_envs_semantics.py

```diff
@@ -0,0 +1,85 @@
+import numpy as np
+import json
+import hashlib
+import os
+
+import logging
+logger = logging.getLogger(__name__)
+
+import gym
+from gym import envs, spaces
+
+from generate_json import create_rollout
+
+DATA_DIR = os.path.dirname(__file__)
+ROLLOUT_FILE = os.path.join(DATA_DIR, 'rollout.json')
+ROLLOUT_STEPS = 100
+episodes = ROLLOUT_STEPS
+steps = ROLLOUT_STEPS
+
+if not os.path.isfile(ROLLOUT_FILE): 
+  with open(ROLLOUT_FILE, "w") as outfile:
+    json.dump({}, outfile, indent=2)
+
+def test_env_semantics(spec):
+  with open(ROLLOUT_FILE) as data_file:
+    rollout_dict = json.load(data_file)
+
+  if spec.id not in rollout_dict:
+    added = create_rollout(spec)
```

Comment:
Let's fail the test here rather than creating a rollout on the fly (it makes less sense e.g. on our CI machines)